config/config_loader.go

```diff
@@ -195,20 +195,20 @@ func GetApplicationConfig() *ApplicationConfig {
 
 // GetProviderConfig find the provider config
 // if not found, create new one
-func GetProviderConfig() ProviderConfig {
+func GetProviderConfig() *ProviderConfig {
```

Comment:
are u sure u  need to change the value of the ProviderConfig by its pointer?

Reply:
Be same as other func. GetMetricConfig / GetApplicationConfig